src/config.c

```diff
@@ -260,6 +292,97 @@ void configureVifs(void) {
+            /* Need rebuild, check if interface is new or already exists (check table below).
+                             old: disabled    new: disabled    -> do nothing
+                             old: disabled    new: downstream  -> addVIF(new), joinmcroutergroups
+                             old: disabled    new: upstream    -> addVIF(new)
+                             old: downstream  new: disabled    -> delVIF(old)
+               state table   old: downstream  new: downstream  -> addvif(new,old)
+                             old: downstream  new: upstream    -> delvif(old), addvif(new)
+                             old: upstream    new: disabled    -> clear routes oldvif, delVIF(old)
+                             old: upstream    new: downstream  -> clear routes oldvif, delvif(old)),addvif(new), joinmcroutergroup
+                             old: upstream    new: upstream    -> addvif(new,old)
+            */
+            for ( oDp=RebuildP->S; oDp<RebuildP->E; oDp++ ) if ( ! strcmp (oDp->Name, Dp->Name) ) break;
+            if ( oDp < RebuildP->E ) {
+                switch (oDp->state) {
+                case IF_STATE_DISABLED:
+                    switch (Dp->state) {
+                    case IF_STATE_DISABLED:   {                                                      continue; }
+                    case IF_STATE_DOWNSTREAM: {                                  oDp=NULL;  join=1;  break; }
+                    case IF_STATE_UPSTREAM:   {                                  oDp=NULL;           break; }
+                    }
+                    break;
+                case IF_STATE_DOWNSTREAM:
+                    switch (Dp->state) {
+                    case IF_STATE_DISABLED:   {                    delVIF(oDp);                      continue; }
+                    case IF_STATE_DOWNSTREAM: {                                                      break; }
+                    case IF_STATE_UPSTREAM:   {                    delVIF(oDp);  oDp=NULL;           break; }
+                    }
+                    break;
+                case IF_STATE_UPSTREAM:
+                    switch (Dp->state) {
+                    case IF_STATE_DISABLED:   { clearRoutes(oDp);  delVIF(oDp);                      continue; }
+                    case IF_STATE_DOWNSTREAM: { clearRoutes(oDp);  delVIF(oDp);  oDp=NULL;  join=1;  break; }
+                    case IF_STATE_UPSTREAM:   {                                                      break; }
+                    }
+                    break;
+                }
+                if (Dp->Flags & IFF_LOOPBACK) continue;
+            } else {
+                // New Interface. Only add valid up/downstream vif.
+                if ( (Dp->Flags & IFF_LOOPBACK) || (Dp->state != IF_STATE_DOWNSTREAM && Dp->state != IF_STATE_UPSTREAM) ) continue;
+                oDp=NULL;
+            }
+        }
+        if(Dp->state == IF_STATE_UPSTREAM) {
+            if (upsvifcount < MAX_UPS_VIFS -1)
+            {
```

Comment:
Just style comment: put `{` on previous line and change swap if-else braches so you can compare `upsvifcount` and `MAX_UPS_VIFS` without `-1`:
```c
if (upsvifcount >= MAX_UPS_VIFS) {
/* report error */
} else {
/* OK */
}
```
`+1` and `-1` are lot of times suspicous.

Reply:
Fixed